here-naksha-lib-psql/src/commonMain/kotlin/naksha/psql/PgTable.kt

```diff
@@ -229,10 +229,20 @@ open class PgTable(
         val PARTITION_BY = when (partitionByColumn) {
             // Not partitioned by itself.
             null -> ""
-            // When we partition by ID, we do this using the first byte of the md5 hash of the feature id
+            // When we partition by ID, we do this using the first byte of the md5 hash of the feature id.
             PgColumn.id -> {
                 require(partitionCount in 2..256) { "Invalid partition-count, expect 2 .. 256, found : $partitionCount" }
-                "PARTITION BY RANGE ((get_byte(digest(id,'md5'),0) % $partitionCount))"
+                "PARTITION BY RANGE ((get_byte(digest(${PgColumn.id},'md5'),0) % $partitionCount))"
+            }
+            // The tuple_number contains the partition-number in the top 8-bit (due to big-endian encoding).
+            PgColumn.tuple_number -> {
+                require(partitionCount in 2..256) { "Invalid partition-count, expect 2 .. 256, found : $partitionCount" }
+                "PARTITION BY RANGE ((get_byte(${PgColumn.tuple_number}, 7) % $partitionCount))"
+            }
+            // The store_number contains the partition-number in the lower 8-bit.
+            PgColumn.store_number -> {
+                require(partitionCount in 2..256) { "Invalid partition-count, expect 2 .. 256, found : $partitionCount" }
+                "PARTITION BY RANGE (((${PgColumn.store_number} & 255) % $partitionCount))"
```

Comment:
It's a general partition-related comment, not specifically about this snippet:

I've seen that apart from tags, you have also changed some stuff related to partitioning, like:
- validating partition count
- enhancig check for id-based partitioning

What was the reasoning and is it worth testing?

Reply:
Actually, i just saw that we calculate an MD5 hash for every insert or update in the database and the client, I just wanted to get rid of this. The first idea was to partition by `tuple_number`, because we used this, when loading feature data from database, and if Postgres would be able to directly know from which partition to load the tuple, this would be great. Sadly, PostgresQL does not allow to use generated columns in partition keys, but the same number that is part of `tuple_number` is as well part of `store_number`, so I used this.
Eventually, this prevents that the database has to calculate an MD5 hash for every row modification (the client anyway does this already, why do it twice?).

Reply:
Should have made a separate PR for this.